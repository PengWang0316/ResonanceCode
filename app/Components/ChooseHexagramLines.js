import React from 'react';
import PropTypes from 'prop-types';

import styles from '../styles/ChooseHexagramLines.module.css';

const ChooseHexagramLines = ({
  handleCoinClick, lineNumber, handleCancel, handleSwitchMode
}) => {
  const handleClick = totalNum => {
    handleCoinClick(lineNumber, totalNum, '');
    handleCancel();
  };
  const handleChengingYangClick = _ => handleClick(9);
  const handleChengingYinClick = _ => handleClick(6);
  const handleChengingFixedYangClick = _ => handleClick(7);
  const handleChengingFixedYinClick = _ => handleClick(8);

  return (
    <div className={`${styles.coinPickUpBgDiv}`}>
      <div className={`${styles.coinPickUpWindowDiv}`}>

        <div className="row mb-2">
          <div className="col"><b>Line {(lineNumber * 1) + 1}</b></div>
          <div className="btn-group col" data-toggle="buttons">
            {/** Putting onClick event on lable because FireFox, Edge, and IE's radio
              * input element does not work appropriatly to fire the click event */}
            <label role="button" tabIndex="-1" className="btn btn-outline-secondary btn-sm" htmlFor="coinMode" onClick={handleSwitchMode}>
              <input type="radio" name="options" id="coinMode" /> Coin Mode
            </label>
            <label className="btn btn-secondary btn-sm active" htmlFor="HexagramMode">
              <input type="radio" name="options" id="HexagramMode" defaultChecked /> Line Mode
            </label>
          </div>
        </div>

        <div className="container">

          <div className="row text-center">
            <div className="col">
              <div>
                <div role="button" tabIndex="-3" onClick={handleChengingYangClick} className={`${styles.hexagramContainer}`}>
                  <div>Changing Yang</div>
                  <div>
                    <div className={`${styles.imgLineSideRed}`} /><div className={`${styles.imgLineMiddleRed}`} /><div className={`${styles.imgLineSideRed}`} />
                  </div>
                </div>
              </div>

              <div>
                <div role="button" tabIndex="-2" onClick={handleChengingYinClick} className={`${styles.hexagramContainer}`}>
                  <div>Changing Yin</div>
                  <div>
                    <div className={`${styles.imgLineSideRed}`} /><div className={`${styles.imgLineMiddleBlank}`} /><div className={`${styles.imgLineSideRed}`} />
                  </div>
                </div>
              </div>
            </div>

            <div className="col">
              <div>
                <div role="button" tabIndex="-1" onClick={handleChengingFixedYangClick} className={`${styles.hexagramContainer}`}>
                  <div>Fixed Yang</div>
                  <div>
                    <div className={`${styles.imgLineSide}`} /><div className={`${styles.imgLineMiddle}`} /><div className={`${styles.imgLineSide}`} />
                  </div>
                </div>
              </div>

              <div>
                <div role="button" tabIndex="0" onClick={handleChengingFixedYinClick} className={`${styles.hexagramContainer}`}>
                  <div>Fixed Yin</div>
                  <div>
                    <div className={`${styles.imgLineSide}`} /><div className={`${styles.imgLineMiddleBlank}`} /><div className={`${styles.imgLineSide}`} />
                  </div>
                </div>
              </div>
            </div>

          </div>
        </div>


        <div className="text-right"><button type="button" className="btn btn-info" onClick={handleCancel}>Cancel</button></div>

      </div>
    </div>
  );
};
ChooseHexagramLines.propTypes = {
  handleCoinClick: PropTypes.func.isRequired,
  lineNumber: PropTypes.string.isRequired,
  handleCancel: PropTypes.func.isRequired,
  handleSwitchMode: PropTypes.func.isRequired
};
export default ChooseHexagramLines;
