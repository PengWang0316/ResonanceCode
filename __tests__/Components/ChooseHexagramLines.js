import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ChooseHexagramLines from '../../app/Components/ChooseHexagramLines';

describe('ChooseHexagramLines Test', () => {
  const defaultProps = {
    handleCoinClick: jest.fn(),
    lineNumber: '2',
    handleCancel: jest.fn(),
    handleSwitchMode: jest.fn()
  };
  const getShallowComponent = (props = defaultProps) => shallow(<ChooseHexagramLines {...props} />);

  test('handleSwitchMode', () => {
    const component = getShallowComponent();
    component.find({ htmlFor: 'coinMode' }).simulate('click');
    expect(defaultProps.handleSwitchMode).toHaveBeenCalledTimes(1);
  });

  test('handleChengingYangClick', () => {
    const handleCancel = jest.fn();
    // const handleCoinClick = jest.fn();
    const component = getShallowComponent({ ...defaultProps, handleCancel });
    component.find({ role: 'button' }).at(1).simulate('click');
    expect(handleCancel).toHaveBeenCalledTimes(1);
    expect(defaultProps.handleCoinClick).toHaveBeenLastCalledWith('2', 9, '');
  });

  test('handleChengingYinClick', () => {
    const handleCancel = jest.fn();
    // const handleCoinClick = jest.fn();
    const component = getShallowComponent({ ...defaultProps, handleCancel });
    component.find({ role: 'button' }).at(2).simulate('click');
    expect(handleCancel).toHaveBeenCalledTimes(1);
    expect(defaultProps.handleCoinClick).toHaveBeenLastCalledWith('2', 6, '');
  });

  test('handleChengingFixedYangClick', () => {
    const handleCancel = jest.fn();
    // const handleCoinClick = jest.fn();
    const component = getShallowComponent({ ...defaultProps, handleCancel });
    component.find({ role: 'button' }).at(3).simulate('click');
    expect(handleCancel).toHaveBeenCalledTimes(1);
    expect(defaultProps.handleCoinClick).toHaveBeenLastCalledWith('2', 7, '');
  });

  test('handleChengingFixedYinClick', () => {
    const handleCancel = jest.fn();
    // const handleCoinClick = jest.fn();
    const component = getShallowComponent({ ...defaultProps, handleCancel });
    component.find({ role: 'button' }).at(4).simulate('click');
    expect(handleCancel).toHaveBeenCalledTimes(1);
    expect(defaultProps.handleCoinClick).toHaveBeenLastCalledWith('2', 8, '');
  });

  test('ChooseHexagramLines snapshot', () => expect(renderer.create(<ChooseHexagramLines {...defaultProps} />).toJSON()).toMatchSnapshot());
});
