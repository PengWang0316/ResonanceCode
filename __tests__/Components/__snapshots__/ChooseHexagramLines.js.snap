// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseHexagramLines Test ChooseHexagramLines snapshot 1`] = `
<div
  className="coinPickUpBgDiv"
>
  <div
    className="coinPickUpWindowDiv"
  >
    <div
      className="row mb-2"
    >
      <div
        className="col"
      >
        <b>
          Line 
          3
        </b>
      </div>
      <div
        className="btn-group btn-group-toggle col"
        data-toggle="buttons"
      >
        <label
          className="btn btn-outline-secondary btn-sm"
          htmlFor="coinMode"
          onClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            }
          }
          role="button"
          tabIndex="-1"
        >
          <input
            id="coinMode"
            name="options"
            type="radio"
          />
           Coin Mode
        </label>
        <label
          className="btn btn-secondary btn-sm active"
          htmlFor="HexagramMode"
        >
          <input
            defaultChecked={true}
            id="HexagramMode"
            name="options"
            type="radio"
          />
           Line Mode
        </label>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="row text-center"
      >
        <div
          className="col"
        >
          <div>
            <div
              className="hexagramContainer"
              onClick={[Function]}
              role="button"
              tabIndex="-3"
            >
              <div>
                Changing Yang
              </div>
              <div>
                <div
                  className="imgLineSideRed"
                />
                <div
                  className="imgLineMiddleRed"
                />
                <div
                  className="imgLineSideRed"
                />
              </div>
            </div>
          </div>
          <div>
            <div
              className="hexagramContainer"
              onClick={[Function]}
              role="button"
              tabIndex="-2"
            >
              <div>
                Changing Yin
              </div>
              <div>
                <div
                  className="imgLineSideRed"
                />
                <div
                  className="imgLineMiddleBlank"
                />
                <div
                  className="imgLineSideRed"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div>
            <div
              className="hexagramContainer"
              onClick={[Function]}
              role="button"
              tabIndex="-1"
            >
              <div>
                Fixed Yang
              </div>
              <div>
                <div
                  className="imgLineSide"
                />
                <div
                  className="imgLineMiddle"
                />
                <div
                  className="imgLineSide"
                />
              </div>
            </div>
          </div>
          <div>
            <div
              className="hexagramContainer"
              onClick={[Function]}
              role="button"
              tabIndex="0"
            >
              <div>
                Fixed Yin
              </div>
              <div>
                <div
                  className="imgLineSide"
                />
                <div
                  className="imgLineMiddleBlank"
                />
                <div
                  className="imgLineSide"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="text-right"
    >
      <button
        className="btn btn-info"
        onClick={[MockFunction]}
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
